test/capture_warnings.rb

```diff
@@ -17,17 +17,19 @@ def initialize(fail_on_warnings = true)
     @output = STDOUT
   end
 
-  def execute!
+  def execute!(minitest_run)
     $VERBOSE = true
     $stderr.reopen(stderr_file.path)
-
-    Minitest.after_run do
+    at_exit do
```

Comment:
since we run minitest in `END` now, we can rely on `at_exit` running *after* minitest ends